conf/hmf_data.config

```diff
@@ -0,0 +1,108 @@
+params {
+    hmf_data_paths {
+        '37' {
+            // AMBER
+            heterozygous_sites            = 'dna_pipeline/copy_number/AmberGermlineSites.37.tsv.gz'
+            // COBALT
+            gc_profile                    = 'dna_pipeline/copy_number/GC_profile.1000bp.37.cnp'
+            diploid_bed                   = 'dna_pipeline/copy_number/DiploidRegions.37.bed.gz'
+            // CUPPA
```

Comment:
This is data provided by Hartwig across several separate locations on their public GCP bucket.

To be able to use these reference files in a publicly available workflow I've organised + bundled the files and hosted them on Cloudflare R2, which is an object storage service that doesn't charge for egress. It is conceptually the same as a public AWS S3 bucket or GCP bucket but without the costs associated with downloading the data.

These files also usually change between releases of the Hartwig workflow, so the ability to version and somewhat quickly update is required.

I suppose the reason these aren't in iGenomes is that hasn't been any strong support for it so far - I asked sometime ago where to host this type of data for nf-core pipelines but never had an answer, so I developed this alternate approach. It would be great to start the process to migrate these files to iGenomes though I still suspect there might be some resistance. I'd also have to understand the ability and process to version/update files.

Given the above, I don't think this is something to attempt before the release but I'd love to make an effort to use iGenomes in parallel to the release - would you be able to give some pointers on how to go about this?

With the current set up there is a strong guarantee that the data will remain available. We (UMCCR, Hartwig) have every intention to support and maintain oncoanalyser/hmftools into the future, and that of course includes reliable distribution of reference data.

In terms of the data organisation I have used the following structure

- Harwig reference genomes and indexes
- hmftools resource files
- TSO500 panel-specific files

All of the files referenced above fall under the 'hmftools resource files', and these are distributed under a single tarball. I think this organisation provides the easiest approach to handle/retrieve the data while allowing relatively simple customisation of the locally staged data.

I have also implemented a feature to stage reference data for a given analysis to help users manage that aspect, which is a common scenario as you point out. Some information on this is available [here](https://nf-co.re/oncoanalyser/dev/docs/usage/#staging-reference-data).

With the reference file organisation combined with automated reference data staging, before launching an analysis a user can retrieve all required reference data with

```bash
nextflow run nf-core/oncoanalyser \
  -profile docker \
  --mode wgts \
  --genome GRCh38_hmf \
  --input samplesheet.csv \
  --outdir output/ \
  --prepare_reference_only
```

And then create a config pointing to the local files

```
params {
 
    genomes {
        GRCh38_hmf {
            fasta           = "/path/to/output/reference_data/1.0.0/20240719_1701/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna"
            fai             = "/path/to/output/reference_data/1.0.0/20240719_1701/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.fai"
            dict            = "/path/to/output/reference_data/1.0.0/20240719_1701/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.dict"
            bwamem2_index   = "/path/to/output/reference_data/1.0.0/20240719_1701/bwa-mem2_index/"
            gridss_index    = "/path/to/output/reference_data/1.0.0/20240719_1701/gridss_index/"
            star_index      = "/path/to/output/reference_data/1.0.0/20240719_1701/star_index/"
        }
    }
 
    ref_data_hmf_data_path        = "/path/to/output/reference_data/1.0.0/20240719_1701/hmftools_data/"
    ref_data_panel_data_path      = "/path/to/output/reference_data/1.0.0/20240719_1701/tso500_panel_data/"
    ref_data_virusbreakenddb_path = "/path/to/output/reference_data/1.0.0/20240719_1701/virusbreakenddb/"
}
```

This has been working well enough for users, though happy to hear your thoughts if you had specific suggestions to improve!

Reply:
Marking this as resolved for now, please reopen if you'd like to continue the discussion here.

Reply:
Created an issue to continue this piece of work:

- https://github.com/nf-core/oncoanalyser/issues/80